`pm2 startup` does not work on macOS – "Failure when trying to write startup script"

Running `pm2 startup` and then running the generated command results in the following error.

```
[PM2][ERROR] Failure when trying to write startup script
ENOENT: no such file or directory, open '/Users/signum/Library/LaunchAgents/pm2.signum.plist'
```

The issue is that the folder `~/Library/LaunchAgents` does not exist on a fresh macOS install (only verified with the M1 Mac Mini).

## Workaround

Manually create the folder first:

```
mkdir -p ~/Library/LaunchAgents
```

Then run `pm2 startup` as you usually would.

--- a/lib/API/Startup.js
+++ b/lib/API/Startup.js
@@ -275,6 +275,7 @@ module.exports = function(CLI) {
       template = getTemplate('launchd');
       destination = path.join(process.env.HOME, 'Library/LaunchAgents/' + launchd_service_name + '.plist');
       commands = [
+        'mkdir -p ' + path.join(process.env.HOME, 'Library/LaunchAgents'),
         'launchctl load -w ' + destination
       ]
       break;

